File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function maxLevelSum(root: TreeNode | null): number {
    if (!root) return 0

    const queue: TreeNode[] = [root]
    let maxSum: number = -Infinity
    let maxLevel: number = 1
    let currentLevel: number = 1

    while (queue.length > 0) {
        let currentSum: number = 0
        const levelSize: number = queue.length

        for (let i = 0; i < levelSize; i++) {
            const node: TreeNode = queue.shift()
            currentSum += node.val

            if (node.left) queue.push(node.left)
            if (node.right) queue.push(node.right)
        }
        if (currentSum > maxSum) {
            maxSum = currentSum
            maxLevel = currentLevel
        }
        currentLevel++
    }
    return maxLevel
};

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        max_sum = float('-inf')
        result_level = 0
        current_level = 1

        queue = deque([root])

        while queue:
            level_size = len(queue)
            current_level_sum = 0

            for _ in range(level_size):
                node = queue.popleft()
                current_level_sum += node.val

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            if current_level_sum > max_sum:
                max_sum = current_level_sum
                result_level = current_level
            
            current_level += 1

        return result_level